Rhythm/Interpolation.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  Tempo.Interpolation.swift
+//  Rhythm
+//
+//  Created by James Bean on 5/23/17.
+//
+//
+
+import Foundation
+import ArithmeticTools
+
+/// Interpolation between two `Tempo` values.
+///
+/// - TODO: Consider defining multiple interpolation types (linear, exponential, etc.)
+///
+public struct Interpolation {
+
+    // MARK: - Associated Types
+    public enum EasingError: Error {
+        case badValue(String, Double)
```

Comment:
Then, I don't think the message `String` is necessary.

Instead, I could imagine the error being called `outOfRange(Double, ClosedRange<Double>)` with the desired range, or even just `outOfRange(Double)`.